documentation/modules/ROOT/pages/connectors/db2.adoc

```diff
@@ -122,21 +129,225 @@ endif::product[]
+For each change that the snapshot captures, the connector emits a `read` event to the Kafka topic for the captured table.
 
-By default, when a Db2 connector performs a snapshot, it does the following:
+The following workflow lists the steps that {prodname} takes to create a snapshot.
+These steps describe the process for a snapshot when the xref:db2-property-snapshot-mode[`snapshot.mode`] configuration property is set to its default value, which is `initial`.
+You can customize the way that the connector creates snapshots by changing the value of the `snapshot.mode` property.
+If you configure a different snapshot mode, the completes the snapshot by using a modified version of this workflow.
 
-. Determines which tables are in capture mode, and thus must be included in the snapshot. By default, all non-system tables are in capture mode. Connector configuration properties, such as `table.exclude.list` and `table.include.list` let you specify which tables should be in capture mode.
-. Obtains a lock on each of the tables in capture mode. This ensures that no schema changes can occur in those tables during the snapshot.
-The level of the lock is determined by the `snapshot.isolation.mode` connector  configuration property.
-. Reads the highest (most recent) LSN position in the server's transaction log.
-. Captures the schema of all tables that are in capture mode. The connector persists this information in its internal database schema history topic.
-. Optional, releases the locks obtained in step 2. Typically, these locks are held for only a short time.
-. At the LSN position read in step 3, the connector scans the capture mode tables as well as their schemas. During the scan, the connector:
-.. Confirms that the table was created before the start of the snapshot. If it was not, the snapshot skips that table. After the snapshot is complete, and the connector starts emitting change events, the connector produces change events for any tables that were created during the snapshot.
-.. Produces a _read_ event for each row in each table that is in capture mode. All _read_ events contain the same LSN position, which is the LSN position that was obtained in step 3.
-.. Emits each _read_ event to the Kafka topic that has the same name as the table.
-. Records the successful completion of the snapshot in the connector offsets.
+.Default workflow that the {prodname} Db2 connector uses to perform an initial snapshot
+
+1. Establish a connection to the database.
+
+2. Determine which tables are in capture mode and should be included in the snapshot.
+  By default, the connector captures all non-system tables.
+  To have the connector capture a subset of tables or table elements, you can set a number of `include` and `exclude` properties to filter the data, for example, xref:db2-property-table-include-list[`table.include.list`] or xref:db2-property-table-exclude-list[`table.exclude.list`].
```

Comment:
Well, yes, although, actually, it's both right? By default. 
I'll add a few sentences to clarify that initial snapshots capture data for all tables, unless you set up include/exclude lists and captures schema for all tables unless you configure `store.only.captured.tables.ddl` or `store.only.captured.tables.ddl`.

Reply:
> Re: # 1, does the property affect the recording of schema structures both during an initial snapshot and at runtime? I.e., once Debezium starts recording schema data for a table, it continues to track future changes to the schema?

Yes that's correct, both of these settings affect the capture of schema and schema changes across both phases.

> Does the property exist only to enable schema capture for tables or databases that are named in exclude lists?
> ...
> Re: # 2 , I'm not sure what to make of "relevant to database/schema's tables ..." I think that is trying to say that when true, the connector captures schema only for databases or tables from which the connector captures data?


No.

Several databases, such as MySQL, Oracle, and SQL Server allow you to install the database engine once and then you can use that engine to manage multiple logical databases.  In this type of multi-tenancy deployment, we then usually refer to the engine as the "instance" and each logical database as a database.  In MySQL, these databases are also referred to as catalogs, and in Oracle they're referred to as pluggable databases or PDBs.

In databases like Oracle and MySQL, user accounts are shared across these databases within the instance and this means that a connector user may have visibility to select against any number of tables across multiple databases within that instance.  Most often, users are interested in capturing changes for a single databases within the instance, but since these users can access and have visibility to tables in other databases, you may want to prevent the schema history topic from containing references to these other tables in other databases that are of no interest.

For MySQL, a typical deployment configuration may look like:
```
database.include.list=inventory
table.include.list=inventory.tab1,inventory.tab2,inventory.tab2
schema.history.internal.store.only.captured.tables.ddl=false
schema.history.internal.store.only.captured.databases.ddl=true
```

In this example, if there is an `inventory` and `fulfillment` database on the MySQL instance, then the Debezium connector will not capture schema changes from the `fulfillment` database at all because I have specified the restriction on the `databases.ddl` as `true`; however the connector will successfully capture all schema changes related to all tables inside the `inventory` database because that database is included.

Hopefully, that makes that clearer.

> As a reader, it's also a little strange to be reading the Db2, Oracle, or SQL Server connector docs and see the description mention the default value for the MySQL connector. A relic of pulling the content from a shared file, but maybe we can do something to exclude that where it's not needed.

Are you referring to the describe of the database property?

Reply:
Yes, exactly: https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/partials/modules/all-connectors/ref-connector-configuration-database-history-properties.adoc#L53